tests/functional_tests/p2p.py

```diff
@@ -183,6 +226,42 @@ def test_p2p_tx_propagation(self):
             assert len(res.tx_hashes) == 1
             assert res.tx_hashes[0] == txid
 
+        return txid
+
+    def test_p2p_block_propagation(self, daemon2_mempool_txid):
+        print('Testing P2P block propagation')
+        daemon2 = Daemon(idx = 2)
+        daemon3 = Daemon(idx = 3)
+
+        # check precondition: txid in daemon2 mempool
+        res = daemon2.get_transaction_pool_hashes()
+        assert daemon2_mempool_txid in res.tx_hashes
+
+        res = daemon2.generateblocks('42ey1afDFnn4886T7196doS9GPMzexD9gXpsZJDwVjeRVdFCSoHnv7KPbBeGpzJBzHRCAs9UxqeoyFQMYbqSWYTfJJQAWDm', 1)
+        block_height = res.height
+
+        # wait until both are synced
+        loops = 5
+        while True:
+            res2 = daemon2.get_info()
```

Comment:
There's a better way to do this, see: https://github.com/monero-project/monero/pull/9795/files